README.md
<h1 align="center">
     💰 DT Money
</h1>

<h3 align="center">
    An interface to register financial incomes and outcomes.
</h3>

<p align="center">
  <img alt="GitHub language count" src="https://img.shields.io/github/languages/count/henbalmant/dtmoney-react?color=%2304D361">

  <img alt="Repository size" src="https://img.shields.io/github/repo-size/henbalmant/dtmoney-react">

  <a href="https://github.com/henbalmant/dtmoney-react/commits/master">
    <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/henbalmant/dtmoney-react">
  </a>

   <img alt="License" src="https://img.shields.io/badge/license-MIT-brightgreen">
   <a href="https://github.com/henbalmant/dtmoney-react/stargazers">
    <img alt="Stargazers" src="https://img.shields.io/github/stars/henbalmant/dtmoney-react?style=social">
  </a>

  <a href="https://batcave.dev.br">
    <img alt="Made by henbalmant" src="https://img.shields.io/badge/made%20by-henbalmant-%237519C1">
  </a>

  <a href="https://batcave.dev.br/">
    <img alt="Stargazers" src="https://img.shields.io/badge/Batcave-Community-%237159c1?style=flat&logo=ghost">
    </a>

</p>

<h4 align="center">
	Status: 🚀 Done
</h4>

# Summary

<!--ts-->

- [About](#-about)
- [How to execute](#-how-to-run-this-project)
  - [Requirements](#requirements)
  - [Running the application](#-running-the-application)
- [Tech Stack](#-tech-stack)
- [How to contribute](#-how-to-contribute)
- [Author](#-author)
- [License](#-license)
<!--te-->

## 💻 About

💰 DT Money is an interface to register financial incomes and outcomes.

<p align="center">
  <a href="https://dtmoney-react-rust.vercel.app/" target="_blank">:arrow_up_small: Open in Vercel</a>
</p>

![Preview](https://github.com/henbalmant/dtmoney-react/blob/master/Preview.gif)

Project developed inside **Ignite 🔥** offered by [Rocketseat](https://youtu.be/-ZV-_7vNRGw).

---

## 🚀 How to run this project

This project has only a frontend application.

### Requirements

Before you begin, you will need to have the following tools installed on your machine:
[Git](https://git-scm.com) and [Yarn](https://yarnpkg.com/getting-started/install)

Besides that, is good to have an editor to visualize and navigate trough the code like [VSCode](https://code.visualstudio.com/)

#### 🎲 Running the application

```bash

# Clone this repository
$ git clone git@github.com:henbalmant/dtmoney-react.git

# Access the project folder with your terminal/cmd
$ cd dtmoney-react

# Install all dependencies
$ yarn install

# Starts the application
$ yarn start

# The server will start on port:3000 - access http://localhost:3000

```

<p align="center">
  <a href="https://dtmoney-react-rust.vercel.app/" target="_blank">:arrow_up_small: Open in Vercel</a>
</p>

---

## 🛠 Tech Stack

The following tools were used in the construction of the project:

#### **Frontend** ([React](https://reactjs.org/) + [TypeScript](https://www.typescriptlang.org/))

- **[React](https://reactjs.org/)**
- **[Styled Components](https://www.styled-components.com/)**
- **[Polished](https://polished.js.org/)**
- **[MirageJS](https://miragejs.com/)**
- **[Axios](https://axios-http.com/ptbr/docs/intro)**

> See [package.json](https://github.com/henbalmant/dtmoney-react/blob/master/package.json)

#### **Utilities**

- Editor: **[Visual Studio Code](https://code.visualstudio.com/)**
- Markdown: **[StackEdit](https://stackedit.io/)**, **[Markdown Emoji](https://gist.github.com/rxaviers/7360908)**
- Commit Conventional: **[Commitlint](https://github.com/conventional-changelog/commitlint)**

---

## 💪 How to contribute

1. **Fork** the project.
2. Create a new branch with your changes: `git checkout -b my-feature`
3. Save your changes and create a commit message telling you what you did: `git commit -m "feature: My new feature"`
4. Submit your changes: `git push origin my-feature`
   > If you have any questions check this guide on [how to contribute](./CONTRIBUTING.md)

---

## 🦸 Author

<a href="https://batcave.dev.br/author/henbalmant/">
 <img style="border-radius: 50%;" src="https://avatars.githubusercontent.com/u/20211646?v=4" width="100px;" alt=""/>
 <br />
 <sub><b>Henrique Balmant</b></sub></a> <a href="https://batcave.dev.br/author/henbalmant/" title="Batcave">🚀</a>
 <br />

[![Gmail Badge](https://img.shields.io/badge/-henrique.balmant@gmail.com-c14438?style=flat-square&logo=Gmail&logoColor=white&link=mailto:henrique.balmant@gmail.com)](mailto:henrique.balmant@gmail.com)
[![Linkedin Badge](https://img.shields.io/badge/-Henrique%20Balmant-blue?style=flat-square&logo=Linkedin&logoColor=white&link=https://www.linkedin.com/in/henrique-balmant/)](https://www.linkedin.com/in/henrique-balmant/)

---

## 📝 License

This project is under the license [MIT](./LICENSE).

Made with ❤️ by Henrique Balmant 👋🏽 [Get in touch!](https://www.linkedin.com/in/henrique-balmant/)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.0.3] - 2022-05-16

### Added

- New transaction modal
- Global modal styling
- Income & outcome buttons
- API mock with MirageJS
- useContext to input and get transactions data
- Summary feature
- useTransactions hook

### Changed

### Removed

### Fixed

## [0.0.2] - 2022-05-16

### Added

- Global styles with styled-components
- Header component
- Dashboard component
- TransactionsTable component

### Changed

### Removed

### Fixed

## [0.0.1] - 2022-05-13

### Added

- create-react-app
- Application assets (icons & logos)

### Changed

### Removed

### Fixed


src/styles/global.ts
import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  :root {
    --background: #f0f2f5;
    --red: #E52e54;
    --blue: #5429CC;
    --green: #33CC95;
    --blue-light: #6933FF;
    --text-title: #363f5f;
    --text-body: #969cb3;
    --shape: #FFFFFF
  }

  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    @media (max-width: 1080px) {
      font-size: 93.75%;
    }

    @media (max-width: 720px) {
      font-size: 87.5%;
    }
  }

  body {
    background: var(--background);
    -webkit-font-smoothing: antialiased;
  }

  body, input, textarea, button {
    font-family: 'Poppins', sans-serif;
    font-weight: 400;
  }

  h1, h2, h3, h4, h5, h6, strong {
    font-weight: 600;
  }

  button {
    cursor: pointer;
  }

  [disabled] {
    opacity: 0.6;
    cursor: not-allowed;
  }

  .react-modal-overlay {
    background: rgba(0,0,0, 0.5);

    position: fixed;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;

    display: flex;
    align-items: center;
    justify-content: center;
  }

  .react-modal-content {
    width: 100%;
    max-width: 576px;
    background: var(--background);
    padding: 3rem;
    position: relative;
    border-radius: 0.25rem;
  }

  .react-modal-close {
    position: absolute;
    right: 1.5rem;
    top: 1.5rem;
    border: 0;
    background: transparent;

    transition: filter 0.2s;

    &:hover {
      filter: brightness(0.9)
    }
  }
`;


src/components/NewTransactionModal/styles.ts
import styled from "styled-components";
import { darken, transparentize } from "polished";

export const Container = styled.form`
  h2 {
    color: var(--text-title);
    font-size: 1.5rem;
    margin-bottom: 2rem;
  }

  input {
    width: 100%;
    padding: 0 1.5rem;
    height: 4rem;
    border-radius: 0.25rem;

    border: 1px solid #d7d7d7;
    background: #e7e9ee;

    font-weight: 400;
    font-size: 1rem;

    &::placeholder {
      color: var(--text-body);
    }

    & + input {
      margin-top: 1rem;
    }
  }

  button[type="submit"] {
    width: 100%;
    padding: 0 1.5rem;
    height: 4rem;
    background: var(--green);
    color: #fff;
    border-radius: 0.25rem;
    border: 0;
    font-size: 1rem;
    font-weight: 600;
    margin-top: 1.5rem;

    transition: filter 0.2s;

    &:hover {
      filter: brightness(0.9);
    }
  }
`;

export const TransactionTypeContainer = styled.div`
  margin: 1rem 0;
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 0.5rem;
`;

interface RadioBoxProps {
  isActive: boolean;
  activeColor: "green" | "red";
}

const colors = {
  green: "#33CC95",
  red: "#E52e54",
};

export const RadioBox = styled.button<RadioBoxProps>`
  height: 4rem;
  border: 1px solid #d7d7d7;
  border-radius: 0.5rem;

  background: ${(props) =>
    props.isActive === true
      ? transparentize(0.9, colors[props.activeColor])
      : "transparent"};

  display: flex;
  align-items: center;
  justify-content: center;

  transition: border-color 0.2s;

  &:hover {
    border-color: ${darken(0.1, "#d7d7d7")};
  }

  img {
    width: 20px;
    height: 20px;
  }

  span {
    display: inline-block;
    margin-left: 1rem;
    font-size: 1rem;
    color: var(--text-title);
  }
`;
